platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/module/telemetry/PerformanceEventTest.java

```diff
@@ -0,0 +1,133 @@
+package com.mapbox.mapboxsdk.module.telemetry;
+
+import android.app.ActivityManager;
+import android.content.Context;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Parcel;
+import android.support.test.runner.AndroidJUnit4;
+import android.util.DisplayMetrics;
+import android.view.Display;
+import android.view.WindowManager;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.mapbox.mapboxsdk.Mapbox;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.UUID;
+
+import static org.junit.Assert.assertNotNull;
+
+@RunWith(AndroidJUnit4.class)
+public class PerformanceEventTest {
+
+
+  @Test
+  public void checksPerformanceEventWithMetaData() throws Exception {
+
+    PerformanceEvent event = obtainPerformanceEvent();
+    assertNotNull(event);
+
+    Parcel parcel = Parcel.obtain();
+
+    event.writeToParcel(parcel, 0);
+    parcel.setDataPosition(0);
+
+    PerformanceEvent newPerfEvent = PerformanceEvent.CREATOR.createFromParcel(parcel);
+    assertNotNull(newPerfEvent);
+
+  }
+
+  @Test
+  public void checksPerformanceEventOnlyRequiredData() throws Exception {
+
+    Bundle bundle = new Bundle();
+    bundle.putString("property ignored", "value will be ignored");
+    PerformanceEvent event = new PerformanceEvent(UUID.randomUUID().toString(), bundle);
+    assertNotNull(event);
+
+    Parcel parcel = Parcel.obtain();
+    event.writeToParcel(parcel, 0);
+    parcel.setDataPosition(0);
+
+    PerformanceEvent newPerfEvent = PerformanceEvent.CREATOR.createFromParcel(parcel);
+    assertNotNull(newPerfEvent);
```

Comment:
This test only validates if createFromParcel doesn't return null while the test name indicates values inputted are being ignored